vscode/scripts/release-notes.ts

```
@@ -0,0 +1,162 @@
+
+    ${summary instanceof Error ? "" : `${summary}`}
+    `
+
+    output += `${intro}\n\n`
+
+    const outro = dedent`
+      **Full Comparison**: https://github.com/sourcegraph/cody/compare/M${previousMinor}...M${minor}
+    `
+    output += `\n${outro}\n`
+    console.log('\n=== Preview of Release Notes ===\n')
+    console.log(output)
+    console.log('\n==============================\n')    
+    // this is saved in the runner's local file system to be used in the release notes
+    await fs.promises.writeFile(path.join(__dirname, '../GITHUB_CHANGELOG.md'), output)
+}
+
+main().catch(console.error)
+
+// Extract a list of changes and the previous version number
+function extractLatestChangelog(changelog: string, currentVersion: string): {content: string, previousVersion: string} {
+    const lines = changelog.split('\n')
+    const changes = []
+    let found = false
+    let previousVersion = ''
+    for (const line of lines) {
+        if (found) {
+            // If previous version header found, stop appending changelog content
+            if (line.startsWith("## ")) {
+                const versionMatches = /^## (?<dottedVersion>\d+\.\d+\.\d+)$/.exec(line)
+                if (!versionMatches?.groups) {
+                    throw new Error(`Malformed version line: ${line}`)
+                }
+                previousVersion = versionMatches.groups?.dottedVersion
+                break
+            }
+            changes.push(line)
+        } else if (line.startsWith(`## ${currentVersion}`)) {
+            found = true
+            changes.push(line)
+        }
+    }
+    return {
+        content: changes.join('\n'),
+        previousVersion
+    }
+}
+
+function extractPreviousMinor(minor: string): string {
+    return `${Number.parseInt(minor) - 2}`
+}
+
+async function summarizeChangelog(changelog: string): Promise<string | Error> {
+    
+    const anthropic = new Anthropic({
```

Can we raise a error message if the `ANTHROPIC_API_KEY` if not already set ?

---

there is a try catch in the anthropic client call, so it'll fail with an error message from anthropic if the key is bad, but i can add some more verbosity